src/commands/generate.js

```diff
@@ -1,154 +1,19 @@
-  }
-
-  // Get the output string and destination filename
-  let template;
-  try {
-    template = fs.readFileSync(path.resolve(__dirname, `../../templates/generate/${command}.js`), {encoding: "utf8"});
-  }
-  catch(e) {
-    return cb("Couldn't read generator file");
-  }
-
-  template = replaceName(template, generatedFileName);
-
-  // Check if the file already exists before we write to it
-  const srcRoot = path.join(CWD, "src");
-  const generateRoot = path.join(srcRoot, availableCommands[command]);
-  const destinationRoot = path.resolve(generateRoot, dirname);
-  const destinationPath = path.join(destinationRoot, generatedFileName + ".js");
-
-  if (destinationPath.indexOf(generateRoot) !== 0) {
-    return cb(`${command} generation is not supported outside of ${path.relative(CWD, generateRoot)}`);
-  }
-
-  let fileExists = true;
-  try {
-    fs.statSync(destinationPath);
-  }
-  catch (e) {
-    fileExists = false;
-  }
-
-  if (fileExists) {
-    return cb(`${filename(destinationPath)} already exists`);
-  }
-
-  // Write output to file
-  fs.mkdirsSync(destinationRoot);
-  fs.writeFileSync(destinationPath, template);
-  logger.success(`New file created: ${filename(destinationPath)}`);
-
-  // If we just generated a reducer, add it to the reducers index
-  if (command === "reducer") {
-    const reducerIndexPath = path.resolve(CWD, "src/reducers/index.js");
-    try {
-      // Get the file contents, but strip off any trailing whitespace. This sets us up
-      // to place the new export on the last line, followed by a blank whitespace line at the end
-      const indexFileContents = fs.readFileSync(reducerIndexPath, {encoding: "utf8"}).replace(/\s*$/, "");
-      const newLine = `export { default as ${generatedFileName} } from "./${generatedFileName}";`;
-
-      // Write back to the index file with the previous contents in addition to our new line and a blank line for git
-      fs.writeFileSync(reducerIndexPath, `${indexFileContents}\n${newLine}\n\n`);
-      logger.success(`${highlight(generatedFileName)} added to reducer index ${filename(reducerIndexPath)}`);
-    }
-    catch (e) {
-      return cb("Unable to modify reducers index. Reducer not added to index");
-    }
-  }
-
-  // Write test file
-  const testFolder = path.resolve(path.join(CWD, "test", availableCommands[command]), dirname);
-
-  // Older generated projects don't have test/reducers or test/containers
-  fs.mkdirsSync(testFolder);
-
-  const testPath = path.join(testFolder, `/${generatedFileName}.test.js`);
-  let testTemplate;
-  let testFileExists = true;
-
-  try {
-    fs.statSync(testPath);
-  }
-  catch (e) {
-    testFileExists = false;
-  }
-
-  if (testFileExists) {
-    return cb(`Unable to create test file for ${highlight(generatedFileName)} because it already exists`);
-  }
+const generator = require("../generator");
 
+module.exports = exports = (generatorName, entityName, options) => {
+  /**
+   * TODO: pass options from CLI to generator
```

Comment:
Should we create another ticket for tackling this or do you think it should block this PR?